--- a/configure
+++ b/configure
@@ -11732,7 +11732,7 @@
 
 
 
-REQ_GTK_VERSION=2.90
+REQ_GTK_VERSION=2.24
 REQ_GLIB_VERSION=2.26.0
 
 pkg_modules="gdk-pixbuf-2.0 glib-2.0 >= $REQ_GLIB_VERSION gio-2.0 >= $REQ_GLIB_VERSION"
@@ -11949,7 +11949,7 @@
 
 fi
 
-tests_modules="gtk+-3.0 >= $REQ_GTK_VERSION"
+tests_modules="gtk+-2.0 >= $REQ_GTK_VERSION"
 # Check whether --enable-tests was given.
 if test "${enable_tests+set}" = set; then :
   enableval=$enable_tests;
