--- a/gtk/Makefile.am
+++ b/gtk/Makefile.am
@@ -1409,34 +1409,7 @@
 	  done;								\
 	  chmod -R u+w stock-icons;					\
         fi &&								\
-	for i in 16 24; do						\
-	   (cd stock-icons/$$i						\
-	   && $(RM) document-save.png 					\
-	   && $(LN_S) media-floppy.png document-save.png 		\
-	   && $(RM) folder-remote.png 					\
-	   && $(LN_S) folder.png folder-remote.png 			\
-	   && $(RM) go-first-rtl.png 					\
-	   && $(LN_S) go-last-ltr.png go-first-rtl.png 			\
-	   && $(RM) go-last-rtl.png 					\
-	   && $(LN_S) go-first-ltr.png go-last-rtl.png 			\
-	   && $(RM) go-next-rtl.png 					\
-	   && $(LN_S) go-previous-ltr.png go-next-rtl.png 		\
-	   && $(RM) go-previous-rtl.png 				\
-	   && $(LN_S) go-next-ltr.png go-previous-rtl.png 		\
-	   && $(RM) media-seek-backward-rtl.png 			\
-	   && $(LN_S) media-seek-forward-ltr.png media-seek-backward-rtl.png \
-	   && $(RM) media-seek-forward-rtl.png 				\
-	   && $(LN_S) media-seek-backward-ltr.png media-seek-forward-rtl.png \
-	   && $(RM) media-skip-backward-rtl.png 			\
-	   && $(LN_S) media-skip-forward-ltr.png media-skip-backward-rtl.png \
-	   && $(RM) media-skip-forward-rtl.png 				\
-	   && $(LN_S) media-skip-backward-ltr.png media-skip-forward-rtl.png \
-	   && $(RM) user-desktop.png 					\
-	   && $(LN_S) folder.png user-desktop.png 			\
-	   && $(RM) user-home.png 					\
-	   && $(LN_S) folder.png user-home.png 				\
-	) done 								\
-	&& touch stamp-icons
+	touch stamp-icons
 
 if CROSS_COMPILING
 gtk_update_icon_cache_program = $(GTK_UPDATE_ICON_CACHE)
