diff --git a/openbox/client.c b/openbox/client.c
index 8317f4b..5559c03 100644
--- a/openbox/client.c
+++ b/openbox/client.c
@@ -3186,12 +3186,12 @@ void client_configure(ObClient *self, gint x, gint y, gint w, gint h,
     /* for app-requested resizes, always resize if 'resized' is true.
        for user-requested ones, only resize if final is true, or when
        resizing in redraw mode */
+    gboolean app_rq = !user && resized;
     send_resize_client = ((!user && resized) ||
                           (user && (final ||
                                     (resized && config_resize_redraw))));
-
     /* if the client is enlarging, then resize the client before the frame */
-    if (send_resize_client && (w > oldclient.width || h > oldclient.height)) {
+    if (!app_rq && send_resize_client && (w > oldclient.width || h > oldclient.height)) {
         XMoveResizeWindow(obt_display, self->window,
                           self->frame->size.left, self->frame->size.top,
                           MAX(w, oldclient.width), MAX(h, oldclient.height));
@@ -3284,7 +3284,8 @@ void client_configure(ObClient *self, gint x, gint y, gint w, gint h,
        both of these resize sections may run, because the top one only resizes
        in the direction that is growing
      */
-    if (send_resize_client && (w <= oldclient.width || h <= oldclient.height))
+    if ((send_resize_client && (w <= oldclient.width || h <= oldclient.height))
+		|| (app_rq && (w != oldclient.width || h != oldclient.height)))
     {
         frame_adjust_client_area(self->frame);
         XMoveResizeWindow(obt_display, self->window,
