--- a/Makefile.am
+++ b/Makefile.am
@@ -6,8 +6,6 @@
 configdir       = $(sysconfdir)/xdg
 rcdir           = $(configdir)/openbox
 xsessionsdir    = $(datadir)/xsessions
-gnomesessiondir = $(datadir)/gnome-session/sessions
-gnomewmfilesdir = $(datadir)/gnome/wm-properties
 pkgconfigdir    = $(libdir)/pkgconfig
 obtpubincludedir= $(includedir)/openbox/@OBT_VERSION@/obt
 rrpubincludedir = $(includedir)/openbox/@RR_VERSION@/obrender
@@ -32,17 +30,13 @@
 
 bin_PROGRAMS = \
 	openbox/openbox \
-	tools/gdm-control/gdm-control \
-	tools/gnome-panel-control/gnome-panel-control \
 	tools/obxprop/obxprop
 
 noinst_PROGRAMS = \
 	obt/obt_unittests
 
 nodist_bin_SCRIPTS = \
-	data/xsession/openbox-session \
-	data/xsession/openbox-gnome-session \
-	data/xsession/openbox-kde-session
+	data/xsession/openbox-session
 
 dist_rc_SCRIPTS = \
 	data/environment
@@ -328,15 +322,6 @@
 	obt/unittest_base.c \
 	obt/bsearch_unittest.c
 
-## gnome-panel-control ##
-
-tools_gnome_panel_control_gnome_panel_control_CPPFLAGS = \
-	$(X_CFLAGS)
-tools_gnome_panel_control_gnome_panel_control_LDADD = \
-	$(X_LIBS)
-tools_gnome_panel_control_gnome_panel_control_SOURCES = \
-	tools/gnome-panel-control/gnome-panel-control.c
-
 ## obxprop ##
 
 tools_obxprop_obxprop_CPPFLAGS = \
@@ -486,9 +471,6 @@
 
 ## data ##
 
-dist_apps_DATA = \
-	data/openbox.desktop
-
 dist_pixmap_DATA = \
 	data/openbox.png
 
@@ -531,18 +513,9 @@
 	@echo make: creating $@
 	@$(edit) $< >$@	
 
-dist_gnomewmfiles_DATA = \
-	data/gnome-wm-properties/openbox.desktop
-
 nodist_xsessions_DATA = \
-	data/xsession/openbox.desktop \
-	data/xsession/openbox-gnome.desktop \
-	data/xsession/openbox-kde.desktop
+	data/xsession/openbox.desktop
 
-dist_gnomesession_DATA = \
-	data/gnome-session/openbox-gnome.session \
-	data/gnome-session/openbox-gnome-fallback.session
-
 dist_noinst_DATA = \
 	data/rc.xsd \
 	data/menu.xsd \
@@ -599,8 +572,6 @@
 nodist_man_MANS = \
 	doc/openbox.1 \
 	doc/openbox-session.1 \
-	doc/openbox-gnome-session.1 \
-	doc/openbox-kde-session.1 \
 	doc/obxprop.1
 
 EXTRA_DIST = \
