SliTaz GNU/Linux patch to Openbox 3.6.1.

--- a/data/xsession/openbox.desktop.in
+++ b/data/xsession/openbox.desktop.in
@@ -1,7 +1,9 @@
 [Desktop Entry]
-Name=Openbox
+Type=XSession
+Name=Openbox Session
+Name[pt]=Sessão Openbox
+Name[ru]=Сеанс OpenBox
 Comment=Log in using the Openbox window manager (without a session manager)
 Exec=@bindir@/openbox-session
 TryExec=@bindir@/openbox-session
 Icon=openbox
-Type=Application
