--- a/tools/musl-gcc.specs.sh
+++ b/tools/musl-gcc.specs.sh
@@ -5,10 +5,10 @@
 %rename cpp_options old_cpp_options
 
 *cpp_options:
--nostdinc -isystem $incdir -isystem include%s %(old_cpp_options)
+-nostdinc -isystem $incdir -isystem /usr/include -isystem include%s %(old_cpp_options)
 
 *cc1:
-%(cc1_cpu) -nostdinc -isystem $incdir -isystem include%s
+%(cc1_cpu) -nostdinc -isystem $incdir -isystem /usr/include -isystem include%s
 
 *link_libgcc:
 -L$libdir -L .%s
