--- a/src/CTPP2FileSourceLoader.cpp
+++ b/src/CTPP2FileSourceLoader.cpp
@@ -38,6 +38,7 @@
 #include <errno.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <unistd.h>
 
 #ifdef WIN32
     #include <direct.h> /* getcwd */
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -809,7 +809,7 @@
               man/ctpp2i.1
               man/ctpp2json.1
               man/ctpp2vm.1
-        DESTINATION man/man1
+        DESTINATION share/man/man1
         PERMISSIONS OWNER_READ OWNER_WRITE
                     GROUP_READ
                     WORLD_READ)
