--- a/data/pysol.desktop
+++ b/data/pysol.desktop
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Name=PySol Fan Club Edition
-Exec=pysol.py
+Exec=pysol
 Terminal=false
 Type=Application
 Categories=Game;CardGame;
--- a/scripts/cardset_viewer.py
+++ b/scripts/cardset_viewer.py
@@ -7,7 +7,7 @@
 from math import sqrt, sin, cos, pi
 from Tkinter import *
 try:
-    import Image, ImageTk
+    from PIL import Image, ImageTk
 except ImportError:
     Image = None
 
--- a/pysollib/mfxutil.py
+++ b/pysollib/mfxutil.py
@@ -48,11 +48,11 @@
         Image = None
     else:
         # for py2exe
-        import GifImagePlugin
-        import PngImagePlugin
-        import JpegImagePlugin
-        import BmpImagePlugin
-        import PpmImagePlugin
+        from PIL import GifImagePlugin
+        from PIL import PngImagePlugin
+        from PIL import JpegImagePlugin
+        from PIL import BmpImagePlugin
+        from PIL import PpmImagePlugin
         Image._initialized = 2
 
 
