--- a/third_party/libwebm/mkvparser/mkvparser.cc
+++ b/third_party/libwebm/mkvparser/mkvparser.cc
@@ -36,8 +36,8 @@
 inline bool isnan(double val) { return !!_isnan(val); }
 inline bool isinf(double val) { return !_finite(val); }
 #else
-inline bool isnan(double val) { return std::isnan(val); }
-inline bool isinf(double val) { return std::isinf(val); }
+inline bool isnan(double val) { return isnan(val); }
+inline bool isinf(double val) { return isinf(val); }
 #endif  // MSC_COMPAT
 
 IMkvReader::~IMkvReader() {}
