--- a/src/Win_GParted.cc
+++ b/src/Win_GParted.cc
@@ -2775,12 +2775,12 @@
 	switch ( action )
 	{
 		case SWAPOFF:
-			cmd = "swapoff -v " + Glib::shell_quote( filesystem_ptn.get_path() );
+			cmd = "swapoff " + Glib::shell_quote( filesystem_ptn.get_path() );
 			success = ! Utils::execute_command( cmd, output, error );
 			error_msg = "<i># " + cmd + "\n" + error + "</i>";
 			break;
 		case SWAPON:
-			cmd = "swapon -v " + Glib::shell_quote( filesystem_ptn.get_path() );
+			cmd = "swapon " + Glib::shell_quote( filesystem_ptn.get_path() );
 			success = ! Utils::execute_command( cmd, output, error );
 			error_msg = "<i># " + cmd + "\n" + error + "</i>";
 			break;
--- a/gparted.in
+++ b/gparted.in
@@ -30,7 +30,7 @@
 #
 #  Only permit one instance of GParted to execute at a time
 #
-if test "z`ps -e | grep gpartedbin`" != "z"; then
+if test "z`pgrep gpartedbin`" != "z"; then
 	echo "The process gpartedbin is already running."
 	echo "Only one gpartedbin process is permitted."
 	exit 1
@@ -94,7 +94,7 @@
 HAVE_SYSTEMCTL=no
 for k in '' `echo "$PATH" | sed 's,:, ,g'`; do
 	if test -x "$k/systemctl"; then
-		if test "z`ps -e | grep systemd`" != "z"; then
+		if test "z`pgrep systemd`" != "z"; then
 			HAVE_SYSTEMCTL=yes
 			break
 		fi
@@ -107,7 +107,7 @@
 #
 HAVE_UDISKS2_INHIBIT=no
 if test -x "/usr/lib/udisks2/udisks2-inhibit"; then
-	if test "z`ps -e | grep 'udisksd'`" != "z"; then
+	if test "z`pgrep 'udisksd'`" != "z"; then
 		HAVE_UDISKS2_INHIBIT=yes
 	fi
 fi
@@ -119,7 +119,7 @@
 HAVE_UDISKS=no
 for k in '' `echo "$PATH" | sed 's,:, ,g'`; do
 	if test -x "$k/udisks"; then
-		if test "z`ps -e | grep udisks-daemon`" != "z"; then
+		if test "z`pgrep udisks-daemon`" != "z"; then
 			HAVE_UDISKS=yes
 			break
 		fi
@@ -133,7 +133,7 @@
 HAVE_HAL_LOCK=no
 for k in '' `echo "$PATH" | sed 's,:, ,g'`; do
 	if test -x "$k/hal-lock"; then
-		if test "z`ps -e | grep hald`" != "z"; then
+		if test "z`pgrep hald`" != "z"; then
 			HAVE_HAL_LOCK=yes
 			break
 		fi
