--- a/include/c99_math.h
+++ b/include/c99_math.h
@@ -192,19 +192,6 @@
  */
 #if __cplusplus >= 201103L && (__GLIBC__ > 2 || (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 23))
 #include <cmath>
-
-using std::fpclassify;
-using std::isfinite;
-using std::isinf;
-using std::isnan;
-using std::isnormal;
-using std::signbit;
-using std::isgreater;
-using std::isgreaterequal;
-using std::isless;
-using std::islessequal;
-using std::islessgreater;
-using std::isunordered;
 #endif
 
 
--- a/src/gallium/drivers/swr/rasterizer/memory/Convert.h
+++ b/src/gallium/drivers/swr/rasterizer/memory/Convert.h
@@ -169,13 +169,13 @@
     mant = uf & 0x007FFFFF;
 
     // Check for out of range
-    if (std::isnan(val))
+    if (isnan(val))
     {
         exp = 0x1F;
         mant = 0x200;
         sign = 1;                     // set the sign bit for NANs
     }
-    else if (std::isinf(val))
+    else if (isinf(val))
     {
         exp = 0x1f;
         mant = 0x0;
--- a/src/gallium/drivers/swr/rasterizer/jitter/builder_misc.cpp
+++ b/src/gallium/drivers/swr/rasterizer/jitter/builder_misc.cpp
@@ -54,13 +54,13 @@
         mant = uf & 0x007FFFFF;
 
         // Check for out of range
-        if (std::isnan(val))
+        if (isnan(val))
         {
             exp = 0x1F;
             mant = 0x200;
             sign = 1;                     // set the sign bit for NANs
         }
-        else if (std::isinf(val))
+        else if (isinf(val))
         {
             exp = 0x1f;
             mant = 0x0;
