--- a/wscript
+++ b/wscript
@@ -433,10 +433,10 @@
         #bld.add_subdirs('tests')
 
     if bld.env['BUILD_DOXYGEN_DOCS'] == True:
-        html_docs_source_dir = "build/default/html"
+        html_docs_source_dir = "html"
         if bld.cmd == 'install':
-            share_dir = bld.options.destdir + bld.env['PREFIX'] + '/share/jack-audio-connection-kit'
-            html_docs_install_dir = share_dir + '/reference/html/'
+            doc_dir = bld.options.destdir + bld.env['PREFIX'] + '/share/doc/jack-audio-connection-kit-' + VERSION
+            html_docs_install_dir = doc_dir + '/reference/html/'
             if os.path.isdir(html_docs_install_dir):
                 Logs.pprint('CYAN', "Removing old doxygen documentation installation...")
                 shutil.rmtree(html_docs_install_dir)
