--- /dev/null
+++ b/doc/apache.conf
@@ -0,0 +1,13 @@
+AliasMatch  ^/zonecheck/?$         /usr/share/zonecheck-cgi/html/form.html
+AliasMatch  ^/zonecheck/(en|fr)/?$ /usr/share/zonecheck-cgi/html/form.html.$1
+AliasMatch  ^/zonecheck/(en|fr)/(.*)$ /usr/share/zonecheck-cgi/html/$2.$1
+ScriptAlias /zonecheck/cgi-bin/zc.cgi /usr/share/zonecheck/zc/zc.rb
+ScriptAlias /zonecheck/cgi-bin/    /usr/share/zonecheck/zc/
+Alias       /zonecheck/js/         /usr/share/zonecheck-cgi/js/
+Alias       /zonecheck/style/      /usr/share/zonecheck-cgi/style/
+Alias       /zonecheck/img/        /usr/share/zonecheck-cgi/img/
+Alias       /zonecheck/            /usr/share/zonecheck-cgi/html/
+
+<Directory /usr/share/zonecheck-cgi/>
+  Options Includes FollowSymLinks MultiViews
+</Directory>
