--- a/src/hax65816.c
+++ b/src/hax65816.c
@@ -209,6 +209,7 @@
                             q = 2;
                             break;
                         }
+                        /* FALLTHROUGH */
                     default:
                         unfetchbyte(n[1]);
                         q = 1;
@@ -232,6 +233,7 @@
                             q = 1;
                             break;
                         }
+                        /* FALLTHROUGH */
                     default:
                         unfetchbyte(n[2]);
                         q = 2;
