--- a/foobillardplus.desktop
+++ b/foobillardplus.desktop
@@ -3,10 +3,10 @@
 Version=1.0
 Type=Application
 Terminal=false
-Icon=/opt/foobillardplus/foobillardplus.png
+Icon=foobillardplus
 Name=Foobillard++
-Exec=/opt/foobillardplus/bin/foobillardplus
+Exec=foobillardplus
 Comment=OpenGL Billiard Game
-Name=Foobillard++
-Comment=OpenGL Billiard Game
+Comment[ru]=Бильярд на OpenGL
 GenericName=OpenGL Billiard Game
+Categories=Game;SportsGame;
