--- a/src/vmath.c
+++ b/src/vmath.c
@@ -52,7 +52,7 @@
 /***********************************************************************
  *            fast sinus implementation lookup table                   *
  ***********************************************************************/
-inline float fastsin(float n) {
+extern inline float fastsin(float n) {
    float f = n * HALF_MAX_CIRCLE_ANGLE / M_PI;
    int i;
    i = (int)f;
@@ -67,7 +67,7 @@
  *            fast cosinus implementation lookup table                 *
  ***********************************************************************/
 
-inline float fastcos(float n) {
+extern inline float fastcos(float n) {
    float f = n * HALF_MAX_CIRCLE_ANGLE / M_PI;
    int i;
    i = (int)f;
@@ -82,7 +82,7 @@
  *                     fast atan implementation                       *
  ***********************************************************************/
 
-inline float fastatan(float x)
+extern inline float fastatan(float x)
 {
     return M_PI_4*x - x*(fabs(x) - 1)*(0.2447 + 0.0663*fabs(x));
 }
@@ -92,7 +92,7 @@
  *                     fast atan2 implementation                       *
  ***********************************************************************/
 
-inline float fastatan2(float y, float x) {
+extern inline float fastatan2(float y, float x) {
 	  float coeff_1 = M_PI / 4.0f;
 	  float coeff_2 = 3.0f * coeff_1;
 	  float abs_y = abs(y);
