Index: magicpoint-1.13a/contrib/tex2eps.sh
===================================================================
--- magicpoint-1.13a.orig/contrib/tex2eps.sh	2008-12-24 11:59:36.000000000 -0500
+++ magicpoint-1.13a/contrib/tex2eps.sh	2008-12-24 12:00:07.000000000 -0500
@@ -13,7 +13,7 @@
 # temporary filename (without .eps suffix)
 tmp=$1
 
-echo '\nopagenumbers' > $tmp.tex
+printf '\\nopagenumbers\n' > $tmp.tex
 cat >> $tmp.tex
 echo '\end' >> $tmp.tex
 tex $tmp.tex > /dev/null 2> /dev/null
