Index: kit/mgp.c
===================================================================
--- kit.orig/mgp.c	2009-02-22 11:29:35.000000000 -0500
+++ kit/mgp.c	2009-02-22 11:31:10.000000000 -0500
@@ -64,7 +64,7 @@
 	"red", "green", "blue", "yellow", "black", "white",
 };
 static char *rakugaki_backcolors[] = {
-	"black", "black", "black", "black", "gray", "gray",
+	"red", "green", "blue", "yellow", "black", "white",
 };
 static int demointerval = 0;	/* XXX define option for this */
 
@@ -783,7 +783,7 @@
 	state.width = window_width;
 	state.height = window_height;
 	if (!pen_curs) {
-		pen_curs = XCreateFontCursor(display, XC_pencil);
+		pen_curs = XCreateFontCursor(display, XC_dot);
 		rakugaki_updatecolor(pen_curs);
 	}
 	state_goto(&state, start_page, 0);
