--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -1,7 +1,7 @@
 bin_PROGRAMS = cairo_clock
 transform = s/_/-/g
 
-cairo_clock_LDFLAGS = -Wl, --export-dynamic
+cairo_clock_LDFLAGS = -Wl,--export-dynamic
 
 cairo_clock_LDADD =	\
 	$(GLIB_LIBS)	\
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -226,7 +226,7 @@
 target_alias = @target_alias@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-cairo_clock_LDFLAGS = -Wl, --export-dynamic
+cairo_clock_LDFLAGS = -Wl,--export-dynamic
 cairo_clock_LDADD = \
 	$(GLIB_LIBS)	\
 	$(GTK_LIBS)	\
