--- a/0.7.0/src/parse/parse.c
+++ b/0.7.0/src/parse/parse.c
@@ -2098,8 +2098,8 @@
 			;
 
 		state->substate = Go;
-		/* Fall through */
 	}
+		/* FALLTHROUGH */
 	case Go:
 		while (1) {
 			char want;
@@ -2296,8 +2296,8 @@
 			;
 
 		state->substate = Go;
-		/* Fall through */
 	}
+		/* FALLTHROUGH */
 	case Go:
 		while (1) {
 			char want;
--- a/0.7.0/src/select/arena_hash.h
+++ b/0.7.0/src/select/arena_hash.h
@@ -50,7 +50,9 @@
 	/* Hash any left over bytes */
 	switch (len) {
 	case 3: h ^= data[2] << 16;
+		/* FALLTHROUGH */
 	case 2: h ^= data[1] << 8;
+		/* FALLTHROUGH */
 	case 1: h ^= data[0];
 		h *= m;
 	}
