--- a/src/vzlist.c
+++ b/src/vzlist.c
@@ -255,7 +255,7 @@
 
 /* Sort functions */
 
-inline int check_empty_param(const void *val1, const void *val2)
+extern inline int check_empty_param(const void *val1, const void *val2)
 {
 	if (val1 == NULL && val2 == NULL)
 		return 0;
@@ -662,7 +662,7 @@
 	1 - match
 	0 - do not match
 */
-inline int check_pattern(char *str, char *pat)
+extern inline int check_pattern(char *str, char *pat)
 {
 	if (pat == NULL)
 		return 1;
@@ -748,7 +748,7 @@
 	return;
 }
 
-inline struct Cveinfo *find_ve(int veid)
+extern inline struct Cveinfo *find_ve(int veid)
 {
 	return (struct Cveinfo *) bsearch(&veid, veinfo, n_veinfo,
 			sizeof(struct Cveinfo), id_search_fn);
