--- a/modules/getenv
+++ b/modules/getenv
@@ -69,7 +69,7 @@
 missing_dir  "$root/etc/slitaz/"
 missing_file "$root/etc/slitaz/slitaz.conf" '/etc/slitaz/slitaz.conf'
 missing_file "$root/etc/slitaz/tazpkg.conf" '/etc/slitaz/tazpkg.conf'
-missing_file "$root/etc/slitaz-release"; fill "$root/etc/slitaz-release" 'cooking'
+missing_file "$root/etc/slitaz-release"; fill "$root/etc/slitaz-release" 'next'
 
 # Read configuration
 if [ -n "$root" ]; then
