--- a/applications/tazpanel-pkgs.desktop
+++ b/applications/tazpanel-pkgs.desktop
@@ -1,11 +1,10 @@
 [Desktop Entry]
 Type=Application
-Name=Package Manager
-Name[fr]=Gestionnaire de paquets
-Name[pt]=Gerenciador de Pacotes
-Name[pt_BR]=Gerenciador de Pacotes
-Name[ru]=Менеджер пакетов
+Name=Package Manager web interface
+Name[fr]=Interface web du gestionnaire de paquets
+Name[pt_BR]=Interface da web do gerenciador de pacotes
+Name[ru]=Веб-интерфейс менеджера пакетов
 TryExec=tazweb
-Exec=sh -c 'tazweb root:$$@127.0.0.1:82/user/pkgs.cgi'
+Exec=tazbox su tazpanel pkgs
 Icon=system-software-installer
 Categories=Settings;PackageManager;
