--- a/icu4c/source/i18n/number_decimalquantity.cpp
+++ b/icu4c/source/i18n/number_decimalquantity.cpp
@@ -330,13 +330,13 @@
     setBcdToZero();
     flags = 0;
     // signbit() from <math.h> handles +0.0 vs -0.0
-    if (std::signbit(n) != 0) {
+    if (signbit(n) != 0) {
         flags |= NEGATIVE_FLAG;
         n = -n;
     }
-    if (std::isnan(n) != 0) {
+    if (isnan(n) != 0) {
         flags |= NAN_FLAG;
-    } else if (std::isfinite(n) == 0) {
+    } else if (isfinite(n) == 0) {
         flags |= INFINITY_FLAG;
     } else if (n != 0) {
         _setToDoubleFast(n);
