--- a/Makefile
+++ b/Makefile
@@ -151,7 +151,7 @@
 linux:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
 	LIBS=-lnsl RANLIB=ranlib ARFLAGS=rv AUX_OBJ=weak_symbols.o \
-	NETGROUP="-DNETGROUP" TLI= VSYSLOG= BUGS= \
+	NETGROUP="-DNETGROUP -DUSE_GETDOMAIN" TLI= VSYSLOG= BUGS= \
 	EXTRA_CFLAGS="-DSYS_ERRLIST_DEFINED -DHAVE_STRERROR -DHAVE_WEAKSYMS -DINET6=1 -Dss_family=__ss_family -Dss_len=__ss_len" all
 
 gnu:
