--- a/scripts/mosh
+++ b/scripts/mosh
@@ -347,8 +347,10 @@
   open(STDERR, ">&STDOUT") or die;
 
   my @sshopts = ( '-n' );
+  my @dbopts = ( );
   if ($ssh_pty) {
       push @sshopts, '-tt';
+      push @dbopts, '-t';
   }
 
   my $ssh_connection = "";
@@ -395,8 +397,13 @@
     # proxy to break mysteriously.
     $ENV{ 'SHELL' } = '/bin/sh';
     my $quoted_proxy_command = shell_quote( $0, "--family=$family" );
+    my $host = $userhost; $host = $1 if ($userhost =~ m{@(.*)});
+    my $ssh = join(' ', @ssh);
+    my $port = 22; $port = $1 if ($ssh =~ m{-p *(\d+)});
+    push @dbopts, ('-J', "$quoted_proxy_command --fake-proxy -- $host $port");
     push @sshopts, ( '-S', 'none', '-o', "ProxyCommand=$quoted_proxy_command --fake-proxy -- %h %p" );
   }
+  @sshopts = @dbopts if (`@ssh -V 2>&1` =~ m{Dropbear});
   my @exec_argv = ( @ssh, @sshopts, $userhost, '--', $ssh_connection . "$server " . shell_quote( @server ) );
   exec @exec_argv;
   die "Cannot exec ssh: $!\n";
