--- a/Documentation/DocBook/Makefile
+++ b/Documentation/DocBook/Makefile
@@ -65,10 +65,10 @@
 	find $(obj)/man -name '*.9' | xargs gzip -nf
 
 installmandocs: mandocs
-	mkdir -p /usr/local/man/man9/
+	mkdir -p $(DESTDIR)/usr/share/man/man9/
 	find $(obj)/man -name '*.9.gz' -printf '%h %f\n' | \
 		sort -k 2 -k 1 | uniq -f 1 | sed -e 's: :/:' | \
-		xargs install -m 644 -t /usr/local/man/man9/
+		xargs install -m 644 -t $(DESTDIR)/usr/share/man/man9/
 
 # no-op for the DocBook toolchain
 epubdocs:
