--- a/src/engine/zoomd.c
+++ b/src/engine/zoomd.c
@@ -280,6 +280,7 @@
 #endif
 #undef bpp1
 
+REGISTERS(0)
 static INLINE void fillline(int line)
 {
     register unsigned char *RESTRICT vbuff = cimage.currlines[line];
--- a/src/engine/zoom.c
+++ b/src/engine/zoom.c
@@ -933,6 +933,7 @@
 }
 
 #ifndef USE_i386ASM
+REGISTERS(0)
 static /*INLINE */ void
 moveoldpoints(void /*@unused@ */ *data1,
 	      struct taskinfo /*@unused@ */ *task,
