--- a/doc/C/Makefile.in
+++ b/doc/C/Makefile.in
@@ -346,7 +346,7 @@
 builddir = @builddir@
 datadir = @datadir@
 datarootdir = @datarootdir@
-docdir = $(datadir)/orage/doc/C
+docdir = @docdir@
 dvidir = @dvidir@
 exec_prefix = @exec_prefix@
 host = @host@
--- a/doc/C/images/Makefile.in
+++ b/doc/C/images/Makefile.in
@@ -323,7 +323,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-imagesdir = $(datadir)/orage/doc/C/images
+imagesdir = @docdir@/images
 images_DATA = \
 	default_calendar.png \
 	orage_calendar_edit_menu.png \
