--- Makefile	2009-08-27 18:09:24.000000000 -0400
+++ Makefile	2009-08-27 18:09:49.000000000 -0400
@@ -45,8 +45,8 @@
 
 install: all
 	@echo Installing...
-	$(Q)install -m 644 $(SND_PCM_BIN) ${DESTDIR}/usr/lib/alsa-lib/
-	$(Q)install -m 644 $(SND_CTL_BIN) ${DESTDIR}/usr/lib/alsa-lib/
+	$(Q)install -Dm755 $(SND_PCM_BIN) ${DESTDIR}/usr/lib/alsa-lib/$(SND_PCM_BIN)
+	$(Q)install -Dm755 $(SND_CTL_BIN) ${DESTDIR}/usr/lib/alsa-lib/$(SND_CTL_BIN)
 
 uninstall:
 	@echo Un-installing... 
