--- a/src/rapidsvn_frame.cpp
+++ b/src/rapidsvn_frame.cpp
@@ -869,7 +869,6 @@
                           wxTE_MULTILINE | wxTE_READONLY);
 
   // as much as the widget can stand
-  m->log->SetMaxLength(0);
 
   m->logTracer = new EventTracer(this);
   m->listener.SetTracer(m->logTracer, false);
--- a/src/report_dlg.cpp
+++ b/src/report_dlg.cpp
@@ -46,7 +46,6 @@
                           wxDefaultPosition,
                           wxSize(500, 200),
                           wxTE_MULTILINE | wxTE_READONLY | wxTE_RICH);
-  m_text->SetMaxLength(0);
 
   if (type == ERROR_REPORT)
   {
