--- a/src/add_dialog.c
+++ b/src/add_dialog.c
@@ -216,7 +216,7 @@
 	gtk_widget_set_can_default(add_dialog->cancel_button, TRUE);
 
 	add_dialog->add_button = gtk_button_new();
-	add_dialog->add_image = xa_main_window_find_image("xarchiver-add.png", GTK_ICON_SIZE_SMALL_TOOLBAR);
+	add_dialog->add_image = gtk_image_new_from_icon_name("add-files-to-archive", GTK_ICON_SIZE_SMALL_TOOLBAR);
 	add_dialog->add_hbox = gtk_hbox_new(FALSE, 4);
 	add_dialog->add_label = gtk_label_new_with_mnemonic(_("_Add"));
 
--- a/src/extract_dialog.c
+++ b/src/extract_dialog.c
@@ -418,7 +418,7 @@
 	gtk_widget_set_can_default(cancel_button, TRUE);
 
 	extract_button = gtk_button_new();
-	extract_image = xa_main_window_find_image("xarchiver-extract.png",GTK_ICON_SIZE_SMALL_TOOLBAR);
+	extract_image = gtk_image_new_from_icon_name("extract-archive",GTK_ICON_SIZE_SMALL_TOOLBAR);
 	extract_hbox = gtk_hbox_new(FALSE,4);
 	extract_label = gtk_label_new_with_mnemonic(_("_Extract"));
 
@@ -737,7 +737,7 @@
 	gtk_widget_set_can_default(cancelbutton1, TRUE);
 
 	extract_button = gtk_button_new();
-	extract_image = xa_main_window_find_image("xarchiver-extract.png",GTK_ICON_SIZE_SMALL_TOOLBAR);
+	extract_image = gtk_image_new_from_icon_name("extract-archive",GTK_ICON_SIZE_SMALL_TOOLBAR);
 	extract_hbox = gtk_hbox_new(FALSE,4);
 	extract_label = gtk_label_new_with_mnemonic(_("_Extract"));
 
--- a/src/interface.c
+++ b/src/interface.c
@@ -127,7 +127,7 @@
 	gtk_widget_show(eextract);
 	gtk_container_add(GTK_CONTAINER(xa_popup_menu), eextract);
 
-	image9 =  xa_main_window_find_image ("xarchiver-extract.png",GTK_ICON_SIZE_MENU);
+	image9 = gtk_image_new_from_icon_name("extract-archive",GTK_ICON_SIZE_MENU);
 	gtk_widget_show (image9);
 	gtk_image_menu_item_set_image(GTK_IMAGE_MENU_ITEM(eextract), image9);
 
@@ -684,14 +684,14 @@
 	listing_text = gtk_image_menu_item_new_with_mnemonic (_("_Text file"));
 	gtk_widget_show (listing_text);
 	gtk_container_add (GTK_CONTAINER (listing_submenu),listing_text);
-	tmp_image = gtk_image_new_from_stock ("gtk-justify-fill",GTK_ICON_SIZE_MENU);
+	tmp_image = gtk_image_new_from_icon_name ("text-x-generic",GTK_ICON_SIZE_MENU);
 	gtk_widget_show (tmp_image);
 	gtk_image_menu_item_set_image (GTK_IMAGE_MENU_ITEM (listing_text),tmp_image);
 
 	listing_html = gtk_image_menu_item_new_with_mnemonic (_("_HTML file"));
 	gtk_widget_show (listing_html);
 	gtk_container_add (GTK_CONTAINER (listing_submenu),listing_html);
-	tmp_image =  xa_main_window_find_image ("xarchiver-html.png",GTK_ICON_SIZE_MENU);
+	tmp_image = gtk_image_new_from_icon_name("text-html",GTK_ICON_SIZE_MENU);
 	gtk_widget_show (tmp_image);
 	gtk_image_menu_item_set_image (GTK_IMAGE_MENU_ITEM (listing_html),tmp_image);
 
@@ -752,7 +752,7 @@
 	gtk_container_add (GTK_CONTAINER (menuitem2_menu),addfile);
 	gtk_widget_add_accelerator(addfile, "activate", accel_group, GDK_KEY_d, GDK_CONTROL_MASK, GTK_ACCEL_VISIBLE);
 
-	image2 = xa_main_window_find_image ("xarchiver-add.png",GTK_ICON_SIZE_MENU);
+	image2 = gtk_image_new_from_icon_name ("add-files-to-archive",GTK_ICON_SIZE_MENU);
 	gtk_widget_show (image2);
 	gtk_image_menu_item_set_image (GTK_IMAGE_MENU_ITEM (addfile),image2);
 
@@ -762,7 +762,7 @@
 	gtk_container_add (GTK_CONTAINER (menuitem2_menu),extract_menu);
 	gtk_widget_add_accelerator(extract_menu, "activate", accel_group, GDK_KEY_e, GDK_CONTROL_MASK, GTK_ACCEL_VISIBLE);
 
-	image2 =  xa_main_window_find_image ("xarchiver-extract.png",GTK_ICON_SIZE_MENU);
+	image2 = gtk_image_new_from_icon_name ("extract-archive",GTK_ICON_SIZE_MENU);
 	gtk_widget_show (image2);
 	gtk_image_menu_item_set_image (GTK_IMAGE_MENU_ITEM (extract_menu),image2);
 
@@ -973,7 +973,7 @@
 	gtk_widget_show (separatortoolitem3);
 	gtk_container_add (GTK_CONTAINER (toolbar1),separatortoolitem3);
 
-	tmp_image = xa_main_window_find_image("xarchiver-add.png",GTK_ICON_SIZE_LARGE_TOOLBAR);
+	tmp_image = gtk_image_new_from_icon_name ("add-files-to-archive",GTK_ICON_SIZE_LARGE_TOOLBAR);
 	gtk_widget_show (tmp_image);
 	AddFile_button = (GtkWidget*) gtk_tool_button_new (tmp_image,_("Add"));
 	gtk_widget_set_sensitive (AddFile_button,FALSE);
@@ -982,7 +982,7 @@
 	gtk_container_add (GTK_CONTAINER (toolbar1),AddFile_button);
 	gtk_tool_item_set_tooltip_text(GTK_TOOL_ITEM(AddFile_button), _("Add files"));
 
-	tmp_image = xa_main_window_find_image("xarchiver-extract.png",GTK_ICON_SIZE_LARGE_TOOLBAR);
+	tmp_image = gtk_image_new_from_icon_name("extract-archive",GTK_ICON_SIZE_LARGE_TOOLBAR);
 	gtk_widget_show (tmp_image);
 	Extract_button = (GtkWidget*) gtk_tool_button_new (tmp_image,_("Extract"));
 	gtk_widget_set_sensitive (Extract_button,FALSE);
@@ -1214,7 +1214,7 @@
 	gtk_widget_modify_style(close_button,rcstyle);
 	g_object_unref(rcstyle);
 
-	image = xa_main_window_find_image("xarchiver-close.png", 8);
+	image = gtk_image_new_from_stock("gtk-close", 8);
 	gtk_container_add (GTK_CONTAINER(close_button),image);
 	align = gtk_alignment_new(1.0, 0.5, 0.0, 0.0);
 	gtk_container_add(GTK_CONTAINER(align),close_button);
@@ -1597,7 +1597,7 @@
 	hbox1 = gtk_hbox_new (FALSE,12);
 	gtk_box_pack_start (GTK_BOX (vbox1),hbox1,TRUE,TRUE,0);
 
-	pixbuf = gtk_icon_theme_load_icon(icon_theme,"gnome-mime-application-zip",40,0,NULL);
+	pixbuf = gtk_icon_theme_load_icon(icon_theme,"application-zip",40,0,NULL);
 	if (!pixbuf) pixbuf = gtk_icon_theme_load_icon(icon_theme,"package-x-generic",40,GTK_ICON_LOOKUP_FORCE_SIZE,NULL);
 	icon_pixbuf = gtk_image_new_from_pixbuf(pixbuf);
 	g_object_unref(pixbuf);
--- a/src/pref_dialog.c
+++ b/src/pref_dialog.c
@@ -155,7 +155,7 @@
 
 	prefs_data->prefs_liststore = gtk_list_store_new (3,GDK_TYPE_PIXBUF,G_TYPE_STRING,G_TYPE_UINT);
 	gtk_list_store_append (prefs_data->prefs_liststore,&iter);
-	icon_pixbuf = gtk_icon_theme_load_icon(icon_theme,"gnome-mime-application-zip",40,0,NULL);
+	icon_pixbuf = gtk_icon_theme_load_icon(icon_theme,"application-zip",40,0,NULL);
 	if (!icon_pixbuf) icon_pixbuf = gtk_icon_theme_load_icon(icon_theme,"package-x-generic",32,GTK_ICON_LOOKUP_FORCE_SIZE,NULL);
 	gtk_list_store_set (prefs_data->prefs_liststore, &iter, 0, icon_pixbuf, 1, _("Archive"),2,0,-1);
 	if(icon_pixbuf != NULL)
