--- a/src/capplet/notification-properties.c
+++ b/src/capplet/notification-properties.c
@@ -402,8 +402,7 @@
 
         dialog->preview = notify_notification_new (_("Notification Test"),
                                                    _("Just a test"),
-                                                   "gnome-util",
-                                                   NULL);
+                                                   "gnome-util");
 
         if (!notify_notification_show (dialog->preview, &error)) {
                 char *message;
