--- a/dillorc
+++ b/dillorc
@@ -146,10 +146,12 @@
 # start_page="http://www.dillo.org"
 # start_page="file:/home/jcid/custom_page.html"
 #start_page="about:splash"
+start_page="file:/usr/share/webhome/dillo.html"
 
 # Set the home location
 # home="file:/home/jcid/HomePage/Home.html"
 #home="http://www.dillo.org/"
+home="file:/usr/share/webhome/dillo.html"
 
 # Set the URLs used by the web search dialog.
 # "%s" is replaced with the search keywords separated by '+'.
@@ -157,7 +159,7 @@
 # You can enable multiple search_url strings at once and select from among
 # them at runtime, with the first being the default.
 # (the prefix serves to search from the Location Bar. e.g. "dd dillo image")
-search_url="dd DuckDuckGo (https) https://duckduckgo.com/lite/?kp=-1&q=%s"
+search_url="dd DuckDuckGo (https) https://duckduckgo.com/lite/?kp=-1&q=%s&t=slitaz"
 search_url="Wikipedia http://www.wikipedia.org/w/index.php?search=%s&go=Go"
 search_url="Free Dictionary http://www.thefreedictionary.com/%s"
 search_url="Startpage (https) https://www.startpage.com/do/search?query=%s"
@@ -222,6 +224,7 @@
 # bg_color=gray
 # bg_color=0xd6d6c0
 #bg_color=0xdcd1ba
+bg_color=white
 
 # If your eyes suffer with white backgrounds, change this.
 #allow_white_bg=YES
@@ -281,14 +284,14 @@
 #
 # Gray theme (traditional)
 #
-ui_fg_color=black
-ui_main_bg_color=#c6c6c6
-ui_text_bg_color=#bfdabf
-ui_selection_color=#191970
-ui_button_highlight_color=#a9a9a9
-ui_tab_active_bg_color=#87aca7
-ui_tab_active_fg_color=black
-ui_tab_bg_color=#b7beb7
+#ui_fg_color=black
+#ui_main_bg_color=#c6c6c6
+#ui_text_bg_color=#bfdabf
+#ui_selection_color=#191970
+#ui_button_highlight_color=#a9a9a9
+#ui_tab_active_bg_color=#87aca7
+#ui_tab_active_fg_color=black
+#ui_tab_bg_color=#b7beb7
 
 #
 # Earthly theme:
@@ -313,7 +316,19 @@
 #ui_tab_active_fg_color=#b60907
 #ui_tab_bg_color=#cac682
 
+#
+# SliTaz theme
+#
+ui_fg_color=black
+ui_main_bg_color=#ededed
+ui_text_bg_color=#d9d4ce
+ui_selection_color=#d66018
+ui_button_highlight_color=#a9a9a9
+ui_tab_active_bg_color=#333333
+ui_tab_active_fg_color=white
+ui_tab_bg_color=#dadada
 
+
 # Size of dillo panel
 # tiny   : buttons, location, and progress boxes in one row
 # small  : location in one row, buttons + progress boxes in another
@@ -321,6 +336,7 @@
 # panel_size=tiny
 # panel_size=small
 #panel_size=medium
+panel_size=tiny
 
 #small_icons=NO
 
@@ -330,7 +346,9 @@
 #show_home=YES
 #show_reload=YES
 #show_save=YES
+show_save=NO
 #show_stop=YES
+show_stop=NO
 #show_bookmarks=YES
 #show_tools=YES
 #show_filemenu=YES
@@ -338,7 +356,9 @@
 #show_url=YES
 #show_search=YES
 #show_help=YES
+show_help=NO
 #show_progress_box=YES
+show_progress_box=NO
 
 # Show tooltip popups for the UI
 #show_ui_tooltip=YES
